INT 1851: DDL manipulation

jsmulrow opened the pull request:
To allow for DDL manipulation in Compass, I added 6 methods:
- createCollection
- dropCollection
- createIndex
- dropIndex
- createDatabase
- dropDatabase

Creating a DB wasn't as straightforward as the other operations, and I may end up altering the process as I learn more about what is required for DDL in Compass. Also, to force MongoDB to save a new DB, I had to create a collection in it, which I called "default." There may be a better way to do this, but this was the easiest method I could find.

durran commented:
@jsmulrow Thanks for the pull... The next thing we need are tests. You can follow the patterns set forth in `test/data-service.test.js` and `test/native-client.test.js`. They are run via `npm test`. Please ensure we have tests for each newly added method.

durran commented:
@jsmulrow You're correct - database creation isn't really a thing. At least there isn't a command for it. I would probably remove the `createDatabase` methods for now, since there's no driver or server equilvalent and the data-service is simply an abstraction on the driver. If we do end up having to "fake" a create database operation, it can be in Compass itself in memory - we don't want to make the assumption the user wants any collection added by default.

jsmulrow commented:
@durran Thanks for the review! I removed the `createDatabase` method and added tests for the remaining five functions, trying to follow the existing patterns.

durran commented:
@jsmulrow Looks good! Last thing I would do is 1) remove the version bump in the package.json and then after merging the pull request in 2) `npm version minor`... This will automatically create the version tag for you as well as bumping the version in the package.json. Otherwise, be sure after this commit that you create a tag for the release if you don't follow the `npm version` route: `git tag -a v1.5.0 -m 'Tagging release:.v1.5.0`. The choice is yours, and merge away. :) 👍

jsmulrow commented:
Ok I removed the bump, merged, then ran npm version minor in my local repo. Should I create a pull request for the new version?

durran commented:
@jsmulrow No need for that - you can do it on master and then `git push && git push --tags`... I'll have a look in the morning to double check everything and then publish. Thanks for the work!

jsmulrow commented:
@durran Gotcha. Thanks again!
